core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -0,0 +1,1799 @@
+                           EmptyScriptPubKey))
+            case _: NonStandardScriptPubKey | _: WitnessCommitment =>
+              throw new UnsupportedOperationException(
+                s"$rawSPK is not yet supported")
+          }
+        }
+
+        // Find the conditional leaf with the pubkeys for which sigs are provided
+        // Hashes are used since we only have the pubkey hash in the p2pkh case
+        val sigs = getRecords[PartialSignature](PartialSignatureKeyId)
+        val hashes = sigs.map(sig => CryptoUtil.sha256Hash160(sig.pubKey.bytes))
+        addLeaves(conditional, Vector.empty)
+        val leaves = builder.result()
+
+        val leafOpt = if (hashes.isEmpty) {
+          leaves.find(leaf => leaf._2.isEmpty)
+        } else {
+          leaves.find(leaf => hashes.forall(leaf._2.contains))
+        }
+
+        leafOpt.flatMap {
+          case (path, _, spk) =>
+            val finalizedOpt = finalize(spk)
+            finalizedOpt.map { finalized =>
+              val nestedScriptSig = finalized
+                .getRecords[FinalizedScriptSig](FinalizedScriptSigKeyId)
+                .head
+              val scriptSig =
+                ConditionalScriptSignature(nestedScriptSig.scriptSig, path)
+              wipeAndAdd(scriptSig)
+            }
+        }
+      case locktime: LockTimeScriptPubKey =>
+        finalize(locktime.nestedScriptPubKey)
+      case _: P2PKHScriptPubKey =>
+        collectSigs(required = 1,
+                    sigs => P2PKHScriptSignature(sigs.head._1, sigs.head._2))
+      case _: P2PKScriptPubKey =>
+        collectSigs(required = 1, sigs => P2PKScriptSignature(sigs.head._1))
+      case multiSig: MultiSignatureScriptPubKey =>
+        def generateScriptSig(
+            sigs: Seq[(ECDigitalSignature, ECPublicKey)]): MultiSignatureScriptSignature = {
+          val sortedSigs = sigs
+            .map {
+              case (sig, pubKey) =>
+                (sig, multiSig.publicKeys.indexOf(pubKey))
+            }
+            .sortBy(_._2)
+            .map(_._1)
+
+          MultiSignatureScriptSignature(sortedSigs)
+        }
+
+        collectSigs(required = multiSig.requiredSigs, generateScriptSig)
+      case EmptyScriptPubKey =>
+        val scriptSig = TrivialTrueScriptSignature
+        Some(wipeAndAdd(scriptSig))
+      case _: NonStandardScriptPubKey | _: UnassignedWitnessScriptPubKey |
+          _: WitnessCommitment =>
+        throw new UnsupportedOperationException(
+          s"$spkToSatisfy is not yet supported")
+    }
+  }
+
+  /**
+    * Takes another InputPSBTMap and adds all records that are not contained in this InputPSBTMap
+    * A record's distinctness is determined by its key
+    * @param other InputPSBTMap to be combined with
+    * @return A InputPSBTMap with the combined data of the two InputPSBTMaps
+    */
+  def combine(other: InputPSBTMap): InputPSBTMap = {
+    InputPSBTMap((this.elements ++ other.elements).distinct)
```

Comment:
we should probably ensure they don't conflict right?

Reply:
same with other `combine`s

Reply:
How would they conflict?

Reply:
two different signatures for the same public key, different UTXOs for a given input, etc.
calling `_.distinct` here will not fail on cases like these

Reply:
If that happens they would get an error for making a PSBTMap with the same key.

When I made this I tried doing `e => e.distinctBy(e.key)` however, this function isn't available in older versions